### PR TITLE
feat: legal dialog

### PR DESCRIPTION
**Pull request by samsiegart**
refs: https://github.com/Agoric/agoric-sdk/issues/6491

In draft until we get final confirmation for content/behavior

image

OLD screenshots:

image

image

### CODE REVIEW

File: src/components/ChainConnection.tsx [hunk 1]
```diff
@@ -116,7 +132,7 @@ const ChainConnection = () => {
           'border border-primary group inline-flex items-center rounded-md px-3 py-2 bg-transparent text-base font-medium text-primary',
           !connectionInProgress && !chainConnection && 'hover:bg-gray-100'
         )}
-        onClick={connect}
+        onClick={() => connect()}
```
Comment (turadg) [inline src/components/ChainConnection.tsx:135]: what happens without this change?

File: src/components/TermsDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,117 @@
+import { Dialog, Transition } from '@headlessui/react';
+import clsx from 'clsx';
+import { Fragment, useState } from 'react';
+import { termsAgreedAtom } from 'store/app';
+import { useSetAtom } from 'jotai';
+
+// Increment every time the current terms change.
+export const currentTerms = 0;
```
Comment (turadg) [inline src/components/TermsDialog.tsx:8]: without this it reads like it contains the term data itself. consider `currentTermsIndex`

File: src/store/app.ts [hunk 3]
```diff
@@ -55,6 +55,8 @@ export const networkConfigAtom = atomWithStorage(
   networkConfigs.mainnet
 );
 
+export const termsAgreedAtom = atomWithStorage('terms-agreed', -1);
```
Comment (turadg) [inline src/store/app.ts:58]: elsewhere this reads like a boolean. consider `termsAgreedUponAtom`

File: src/components/TermsDialog.tsx [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+                    quality, availability, accessibility, accuracy or safety. As
+                    more fully explained in the Terms of Use (available here)
+                    and the Risk Statement (available here), your access to and
+                    use of the Inter Protocol is entirely at your own risk and
+                    could lead to substantial losses. You take full
+                    responsibility for your use of the Inter Protocol, and
+                    acknowledge that you use it on the basis of your own
+                    enquiry, without solicitation or inducement by Contributors
+                    (as defined in the Terms of Use). 
+                  </p>
```
Comment (turadg) [inline src/components/TermsDialog.tsx:78]: this should be in a JSON file that PMs could edit. It should also include the history of all terms for transparency about how they've changed.

**Review by turadg — CHANGES_REQUESTED** [review]

File: src/components/ChainConnection.tsx (hunk 1 above)
Comment (samsiegart) [reply src/components/ChainConnection.tsx:135]: It's just a type signature problem, `onClick` is an event handler and I gave `connect` a param which doesn't follow that interface

File: src/components/TermsDialog.tsx (hunk 4 above)
Comment (samsiegart) [reply src/components/TermsDialog.tsx]: Obsolete now that we're linking out to ToS

File: src/store/app.ts (hunk 3 above)
Comment (samsiegart) [reply src/store/app.ts]: changed to `termsIndexAgreedUponAtom`

File: src/components/TermsDialog.tsx (hunk 2 above)
Comment (samsiegart) [reply src/components/TermsDialog.tsx]: done

File: src/components/TermsDialog.tsx [hunk 5]
```diff
@@ -0,0 +1,113 @@
+import { Dialog, Transition } from '@headlessui/react';
+import clsx from 'clsx';
+import { Fragment, useState } from 'react';
+import { termsIndexAgreedUponAtom } from 'store/app';
+import { useSetAtom } from 'jotai';
+
+// Increment every time the current terms change.
```
Comment (turadg) [inline src/components/TermsDialog.tsx:7]: how will we ensure this happens? (in all affected repos)

not a blocker but I think it deserves a ticket for follow-up

**Review by turadg — APPROVED** [review]

**Pull request merged**